lib/board.rb

```diff
@@ -1,32 +1,44 @@
-class GameBoard
-    attr_accessor :positions
-
-    def initialize(positions)
-        @positions = positions
-    end
-
-    def display(positions)
-        puts "\t  #{positions[6]} | #{positions[7]} | #{positions[8]}"
-        puts "\t ---|---|---"
-        puts "\t  #{positions[3]} | #{positions[4]} | #{positions[5]}"
-        puts "\t ---|---|---"
-        puts "\t  #{positions[0]} | #{positions[1]} | #{positions[2]}"
-        puts ""
-    end
-
-     def fill_board(pos, char)
-        @positions[pos] = char
-    end
-
-    def if_full?
-        @positions[0..8].all?{|x| x.instance_of?(String)}
-    end
-
-     def is_fulled?(num)
-        @positions[num] == "X" || "O" == @positions[num]
-    end
-
-end
-
-   
-
+class Board
+  include Interface
+
+  attr_accessor :positions
+
+  def initialize()
+    @positions = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    @lines = [
+          [1,2,3],[4,5,6],[7,8,9],
+          [1,4,7],[2,5,8],[3,6,9],
+          [1,5,9],[3,5,7]
+         ]  
+  end
+
+  def initial
```

Comment:
Guys this method might be redundant/unnecessary. We already are declaring `@positions` in `initialize`.

Reply:
it is needed as we will use it to reset the game when the user need to re-play, already rename the method as "restart".